src/sediment_model.jl

```diff
@@ -13,138 +13,143 @@ function initialize_sediment_model(config::Config)
+        dataset,
+        config,
+        "lateral.land.slope";
+        optional = false,
+        sel = indices_subcatch,
+        type = Float,
+    )
+    clamp!(land_slope, 0.00001, Inf)
+
+    river_length = river_length_2d[indices_river]
+    minimum(river_length) > 0 || error("river length must be positive on river cells")
 
-    riverlength = riverlength_2d[inds_riv]
-    riverwidth = riverwidth_2d[inds_riv]
-    minimum(riverlength) > 0 || error("river length must be positive on river cells")
-    minimum(riverwidth) > 0 || error("river width must be positive on river cells")
+    river_width = river_width_2d[indices_river]
+    minimum(river_width) > 0 || error("river width must be positive on river cells")
 
-    ldd_riv = ldd_2d[inds_riv]
-    graph_riv = flowgraph(ldd_riv, inds_riv, pcr_dir)
+    ldd_river = ldd_2d[indices_river]
+    graph_river = flowgraph(ldd_river, indices_river, pcr_dir)
 
-    index_river = filter(i -> !isequal(river[i], 0), 1:n)
-    frac_toriver = fraction_runoff_toriver(graph, ldd, index_river, landslope, n)
+    index_river = filter(i -> !isequal(river[i], 0), 1:number_of_cells)
+    frac_to_river =
+        fraction_runoff_toriver(graph, ldd, index_river, land_slope, number_of_cells)
 
-    rs = initialize_riversed(nc, config, riverwidth, riverlength, inds_riv)
+    river_sediment =
+        initialize_riversed(dataset, config, river_width, river_length, indices_river)
 
-    modelmap = (vertical = eros, lateral = (land = ols, river = rs))
+    model_map = (
+        vertical = erosion,
+        lateral = (land = overland_flow_sediment, river = river_sediment),
+    )
     indices_reverse = (
-        land = rev_inds,
-        river = rev_inds_riv,
-        reservoir = isempty(reservoir) ? nothing : reservoir.reverse_indices,
-        lake = isempty(lake) ? nothing : lake.reverse_indices,
+        land = reverse_indices_subcatch,
+        river = reverse_indices_river,
+        reservoir = nothing,
+        lake = nothing,
     )
-    writer = prepare_writer(config, modelmap, indices_reverse, x_nc, y_nc, nc)
-    close(nc)
+    writer = prepare_writer(config, model_map, indices_reverse, x_axis, y_axis, dataset)
+    close(dataset)
 
     # for each domain save the directed acyclic graph, the traversion order,
     # and the indices that map it back to the two dimensional grid
     land = (
         graph = graph,
         order = topological_sort_by_dfs(graph),
-        indices = inds,
-        reverse_indices = rev_inds,
+        indices = indices_subcatch,
+        reverse_indices = reverse_indices_subcatch,
     )
     river = (
-        graph = graph_riv,
-        order = topological_sort_by_dfs(graph_riv),
-        indices = inds_riv,
-        reverse_indices = rev_inds_riv,
+        graph = graph_river,
+        order = topological_sort_by_dfs(graph_river),
+        indices = indices_river,
+        reverse_indices = reverse_indices_river,
     )
 
+    sediment_model = SedimentModel()
```

Comment:
What is the advantage of doing this? Why not use directly `SedimentModel()`?

Reply:
It facilitates debugging.
Having the construction of an object as a separate statement instead of nesting its construction in the construction of another object add additional flexibility for placing breakpoints.
For the same reasoning I also refrain from constructing objects during a return statement.